sql/core/src/test/scala/org/apache/spark/sql/DataFrameJoinSuite.scala

```diff
@@ -117,6 +117,57 @@ class DataFrameJoinSuite extends QueryTest with SharedSQLContext {
         .collect().toSeq)
   }
 
+  test("join - self join without manual alias") {
```

Comment:
@cloud-fan Could we add some tests for nested columns ? In case of nested columns, do we need to carry more metadata ?

Reply:
The problem only occurs for `AttributeReference`, which top level columns. I think nested column is totally unrelated here.

Reply:
@cloud-fan Ah.. ok. Thanks..

Reply:
Could you add `SPARK-27547` at the test case name?